lib/hanami/model/config/adapter.rb

```diff
@@ -95,7 +95,7 @@ def load_adapter
           begin
             require "hanami/model/adapters/#{type}_adapter"
           rescue LoadError => e
-            raise LoadError.new("Cannot find Hanami::Model adapter '#{type}' (#{e.message})")
+            raise AdapterNotFound.new(class_name, e.message)
```

Comment:
Leverages Hanami::Model::Error:AdapterNotFound custom error, instead of using LoadError